### PR TITLE
integration: Fix testAddingUserNonFullyConnectedFederation and testNotificationsForOfflineBackends

### PR DESCRIPTION
**Pull request by akshaymankar**
https://wearezeta.atlassian.net/browse/WPB-3974

## Checklist

 - [x] ~Add a new entry in an appropriate subdirectory of `changelog.d`~ No changelog, insignificant change.
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: integration/test/Test/Conversation.hs [hunk 1]
```diff
@@ -410,19 +411,33 @@ testAddUnreachable = do
...
+    own <- asString OwnDomain
+    other <- asString OtherDomain
+
+    -- Ensure that dynamic backend only federates with own domain, but not other
+    -- domain.
+    --
+    -- FUTUREWORK: deleteAllFedConns at the time of acquiring a backend, so
+    -- tests don't affect each other.
+    deleteAllFedConns dynBackend
+    void $ createFedConn dynBackend (FedConn own "full_search")
```
Comment (mdimjasevic) [inline integration/test/Test/Conversation.hs:429]: Can you please explain how this dance of deleting and then immediately making a connection works?

Comment (akshaymankar) [reply integration/test/Test/Conversation.hs:429]: If you have two tests: one of them connects `dynBackend1` with all backends and second one only connects `dynBackend`1 with `OwnDomain`. if the first test runs first, the second test will always fail. So, the second test must always delete all the connections `dynBackend1` has and then only create a connection with `OwnDomain` to ensure that it is not affected by the first test.

Comment (mdimjasevic) [reply integration/test/Test/Conversation.hs:429]: I see.

Wasn't the plan to provide more isolation between test executions? This dance within the test is very fragile, it incurs what I hope can be called a needless brain load, and is not a general solution.

Is there something we can do in the (dynamic) backend setup that would make them isolated from each other (even if we go with e.g. randomly assigning identifiers (names) and such, which should be good enough if they are very unlikely to collide)?

Comment (akshaymankar) [reply integration/test/Test/Conversation.hs:429]: Yes, that's why I added the FUTUREWORK. While claiming an environment we could simply truncate the table which keeps this information around.

File: integration/test/Test/Federation.hs [hunk 2]
```diff
@@ -68,12 +69,13 @@ testNotificationsForOfflineBackends = do
     bindResponse (postConversation delUser (defProteus {qualifiedUsers = [otherUser, downUser1]})) $ \resp ->
       resp.status `shouldMatchInt` 533
 
-    -- Adding users to an up backend conversation should work even when one of
-    -- the participating backends is down
-    otherUser2 <- randomUser OtherDomain def
-    connectUsers delUser otherUser2
-    bindResponse (addMembers delUser upBackendConv [otherUser2]) $ \resp ->
-      resp.status `shouldMatchInt` 200
+    -- Adding users to an up backend conversation should work not work when one
```
Comment (mdimjasevic) [inline integration/test/Test/Federation.hs:72]: "should work not work" - must be a left-over in here, but you tell which is which.

Comment (akshaymankar) [reply integration/test/Test/Federation.hs]: Good catch, It is "should not work".

File: integration/test/Test/Federation.hs [hunk 3]
```diff
@@ -103,23 +108,17 @@ testNotificationsForOfflineBackends = do
...
                 isNotifForUser delUser
               ]
       void $ awaitNotification downUser1 downClient1 (Just newMsgNotif) 1 isDelUserLeaveDownConvNotif
-      void $ awaitNotification otherUser otherClient noValue 1 isDelUserLeaveDownConvNotif
 
       -- FUTUREWORK: Uncomment after fixing this bug: https://wearezeta.atlassian.net/browse/WPB-3664
-      -- void $ awaitNotification downUser1 downClient1 (Just newMsgNotif) 1 (allPreds [isConvLeaveNotif, isNotifConv upBackendConv, isNotifForUser otherUser])
-      -- void $ awaitNotification downUser1 downClient1 (Just newMsgNotif) 1 (allPreds [isConvLeaveNotif, isNotifConv upBackendConv, isNotifForUser delUser])
+      -- void $ awaitNotification downUser1 downClient1 (Just newMsgNotif) 1 isOtherUser2LeaveUpConvNotif
+      -- void $ awaitNotification otherUser otherClient (Just newMsgNotif) 1 isDelUserLeaveDownConvNotif
```
Comment (mdimjasevic) [inline integration/test/Test/Federation.hs:121]: Are you planning to uncomment or remove these two lines?

**Review by mdimjasevic — APPROVED** [review]
Looks good to me, but please do something with the two commented out lines at the bottom of changes in the test module `Federation.hs`.

File: integration/test/Test/Federation.hs (hunk 3 above)
Comment (akshaymankar) [reply integration/test/Test/Federation.hs:121]: Yes, see the comment above them. These don't work due to a bug: https://wearezeta.atlassian.net/browse/WPB-3797

**Pull request merged**